### PR TITLE
Shortcircuit result storage

### PR DESCRIPTION
**Pull request by niconoir**

### CODE REVIEW

**Comment by etiennehomer** [comment]
In the DTO, you add new implementations of Fault, FaultResult, ShortCircuitAnalysisResult etc. Those classes are already implemented in powsybl-core/shortcircuit-api. Do we need our implementations of these classes or we can use the powsybl ones ?

File: src/main/java/org/gridsuite/shortcircuit/server/repositories/ShortCircuitAnalysisResultRepository.java [hunk 1]
```diff
@@ -31,16 +29,33 @@ public class ShortCircuitAnalysisResultRepository {
...
 
-    private static ResultEntity toResultEntity(UUID resultUuid, String result) {
-        return new ResultEntity(resultUuid, result);
+    private static ShortCircuitAnalysisResultEntity toResultEntity(UUID resultUuid, ShortCircuitAnalysisResult result) {
+        List<FaultResultEntity> faultResults = result.getFaultResults().stream().map(ShortCircuitAnalysisResultRepository::toFaultResultEntity).collect(Collectors.toList());
+        return new ShortCircuitAnalysisResultEntity(resultUuid, LocalDateTime.now(), faultResults);
+    }
+
+    private static FaultResultEntity toFaultResultEntity(FaultResult faultResult) {
+        Fault fault = faultResult.getFault();
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/shortcircuit/server/repositories/ShortCircuitAnalysisResultRepository.java:44]: Use shortcircuit.server.dto.Fault here if we keep the shortcircuit-server's dto ?

File: src/main/java/org/gridsuite/shortcircuit/server/service/ShortCircuitService.java [hunk 2]
```diff
@@ -47,8 +49,36 @@ public UUID runAndSaveResult(ShortCircuitRunContext runContext) {
...
+                limitViolationEmbeddable.getLimit(), limitViolationEmbeddable.getLimitName(), limitViolationEmbeddable.getValue());
+    }
+
+    private static FeederResult fromEntity(FeederResultEmbeddable feederResultEmbeddable) {
+        return new FeederResult(feederResultEmbeddable.getConnectableId(), feederResultEmbeddable.getCurrent());
+    }
+
+    public ShortCircuitAnalysisResult getResult(UUID resultUuid, boolean full) {
+        ShortCircuitAnalysisResultEntity result = resultRepository.find(resultUuid);
+        return result != null ? fromEntity(result, full) : null;
```
Comment (flomillot) [inline src/main/java/org/gridsuite/shortcircuit/server/service/ShortCircuitService.java:81]: Maybe if we return an Optional in the repository all the chain will be cleaner ?

**Comment by niconoir** [comment]
> In the DTO, you add new implementations of Fault, FaultResult, ShortCircuitAnalysisResult etc. Those classes are already implemented in powsybl-core/shortcircuit-api. Do we need our implementations of these classes or we can use the powsybl ones ?
> 
> And Fault is only used for results in this PR. But it will be used as parameters for selective studies in the future.

Currently, we do not store the complete results as returned by powsybl. Only informations that are displayed by the frontend are stored, so I chose to use a simpler/lighter implementation for the DTO. Maybe in the future all data returned by powsybl will be needed by the frontend and so maybe the full powsybl implementation for the shortcircuit results might be used and stored in database.

File: src/main/java/org/gridsuite/shortcircuit/server/repositories/ShortCircuitAnalysisResultRepository.java (hunk 1 above)
Comment (niconoir) [reply src/main/java/org/gridsuite/shortcircuit/server/repositories/ShortCircuitAnalysisResultRepository.java:44]: The method is used to convert powsybl shortcircuit results to entities to be stored in database. That's why we use the powsybl version of the results here, and not the dto implementation.

File: src/main/java/org/gridsuite/shortcircuit/server/entities/ShortCircuitAnalysisResultEntity.java [hunk 3]
```diff
@@ -10,26 +10,27 @@
...
+@Table(name = "shortcircuit_result")
+public class ShortCircuitAnalysisResultEntity {
 
     @Id
     private UUID resultUuid;
 
-    @Column(name = "result", columnDefinition = "CLOB")
-    private String result;
+    @Column
+    private LocalDateTime writeTimeStamp;
```
Comment (flomillot) [inline src/main/java/org/gridsuite/shortcircuit/server/entities/ShortCircuitAnalysisResultEntity.java:32]: Is there a reason to take LocalDateTime ? By experience it's better to use ZonedDateTime and specify the time zone.

Comment (niconoir) [reply src/main/java/org/gridsuite/shortcircuit/server/entities/ShortCircuitAnalysisResultEntity.java]: You're right, corrected.

File: src/main/java/org/gridsuite/shortcircuit/server/service/ShortCircuitService.java (hunk 2 above)
Comment (niconoir) [reply src/main/java/org/gridsuite/shortcircuit/server/service/ShortCircuitService.java]: Why not, modified.

**Review by etiennehomer — APPROVED** [review]

**Review by flomillot — APPROVED** [review]

**Pull request merged**